source/onednngraph.js

```diff
@@ -0,0 +1,515 @@
+                        // leave color of fake_backend node by default
+                    }
+                }
+            }
+        }
+
+        const inputs = node.inputs || [];
+        let inputIndex = 0;
+        for (const input of inputs) {
+            if (inputIndex < inputs.length) {
+                const inputArguments = [];
+                const lt = new onednngraph.LogicalTensor(input.id, input.dtype, input.shape, input.stride, input.layout_type, input.property_type);
+                const shape = new onednngraph.TensorShape(lt.shape);
+                const type = new onednngraph.TensorType(lt.dataType, shape);
+                inputArguments.push(arg(input.id.toString(), type, lt));
+                const inputName = (inputs.length == 1) ? 'input' : ('input' + (inputIndex)).toString();
+                this._inputs.push(new onednngraph.Parameter(inputName, inputArguments));
+                inputIndex += 1;
+            }
+        }
+        this._inputs.push(...inputs.slice(inputIndex).map((input, index) => {
+            const inputName = ((inputIndex + index) == 0) ? 'input' : (inputIndex + index).toString();
+            return new onednngraph.Parameter(inputName, true, [arg(input)]);
+        }));
+
+        const outputs = node.outputs || [];
+        let outputIndex = 0;
+        for (const output of outputs) {
+            if (outputIndex < outputs.length) {
+                const outputArguments = [];
+                const lt = new onednngraph.LogicalTensor(output.id, output.dtype, output.shape, output.stride, output.layout_type, output.property_type);
+                const shape = new onednngraph.TensorShape(lt.shape);
+                const type = new onednngraph.TensorType(lt.dataType, shape);
+                outputArguments.push(arg(output.id.toString(), type, lt));
+                const outputName = (outputs.length == 1) ? 'output' : ('output' + (outputIndex)).toString();
+                this._outputs.push(new onednngraph.Parameter(outputName, outputArguments));
+                outputIndex += 1;
+            }
+        }
+        this._outputs.push(...outputs.slice(outputIndex).map((output, index) => {
+            const outputName = ((outputIndex + index) == 0) ? 'output' : (outputIndex + index).toString();
+            return new onednngraph.Parameter(outputName, true, [arg(output)]);
+        }));
+    }
+
+    get type() {
+        return this._type;
+    }
+
+    get name() {
+        return this._name;
+    }
+
+    get inputs() {
+        return this._inputs;
+    }
+
+    get outputs() {
+        return this._outputs;
+    }
+
+    get attributes() {
+        return this._attributes;
+    }
+
+    get description() {
+        return this._description;
+    }
+
+    get device() {
+        return this._device;
+    }
+};
+
+onednngraph.Attribute = class {
+
+    constructor(metadata, type, name, attr_type, attr_value) {
+        this._name = name;
+        this._type = attr_type;
```

Comment:
`type` should return the same types as all other formats. For example, `boolean`, `int32`, `uint32`, `uint32[]`.

Reply:
changed

Reply:
Looks like it would be more straightforward to write the `onndnn.Attribute` constructor to translate instead of the the `switch (schema.type)` code?

Reply:
Is the `switch (schema.type) {` code in `onednn.Attribute` actually required. The format seems to include the attribute name and type in the file so the code transforming attribute values based on types could be merged into `switch (attr_type) {`.